common/hyperscaler/rules/rules_service.go

```diff
@@ -64,13 +68,63 @@ func NewRulesServiceFromSlice(rules []string, enabledPlans *broker.EnablePlans)
 		},
 	}
 
-	rs.ParsedRuleset = rs.parse(rulesConfig)
+	rs.ParsedRuleset = rs.process(rulesConfig)
+	_, rs.ValidRules, rs.ValidationInfo = rs.processAndValidate(rulesConfig)
 	return rs, nil
 }
 
-func (rs *RulesService) parse(rulesConfig *RulesConfig) *ParsingResults {
+func (rs *RulesService) processAndValidate(rulesConfig *RulesConfig) (bool, *ValidRuleset, *ValidationErrors) {
```

Comment:
do we need the bool returned?

Reply:
We do not. It is redundant however more explicit.
Removed in the next push.